Created more complex shortestPath test case to reproduce limitation

craigtaverner opened the pull request.

craigtaverner committed:
Created more complex shortestPath test case to reproduce limitation

pontusmelke commented on community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathLongerAcceptanceTest.scala:
```diff
@@ -0,0 +1,81 @@
+package org.neo4j.internal.cypher.acceptance
```
license

craigtaverner committed:
Added missing license header

craigtaverner committed:
More shortestPath test coverage for investigation of cypher based shortestPath/allshortestPath

craigtaverner committed:
Initial support for shortestPath debug monitoring

cleishm commented:
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;6P5K5kXB&#x2F;494-mergepattern-getorcreate-operator"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> MergePattern (GetOrCreate operator)</a>

craigtaverner committed:
ShortestPath test updates

henriknyman committed:
Move debug information out of shortest path

Exposes a monitor in shortest path (graph algo) that is used in the tests

henriknyman committed:
Plan shortest path with a exhaustive fallback

A fallback rhs branch of an AntiConditionalApply is planned when the shortest path
algorithm can fail to produce results because of complex predicates
- Added optional pathName to VarExpand and path producing to VarLengthExpandPipe.
- Reverted change to ShortestPathExpression return type

WIP: TODOs remaining and debug prints to be cleaned up

craigtaverner committed:
Fix all-shortest-path returning duplicate paths

craigtaverner committed:
ShortestPath with fallback to exhaustive first draft

henriknyman committed:
Fix a bug where allShortestPaths could give wrong results

- Disable treating node predicates as safe (until we support predicates in node expander)
- Add more test coverage

henriknyman committed:
Increase test coverage

- Add test that confirms Fixes #5803
- Confirm when fallback should or should not be executed
- Minor cleanup

Mats-SX commented:
This needs a rebase.

henriknyman commented:
This is already merged as part of this PR https://github.com/neo4j/neo4j/pull/6004
